webapp/channels/src/components/autosize_textarea.tsx

```diff
@@ -13,203 +14,169 @@ type Props = {
-            placeholder,
-            disabled,
-            onInput,
-
-            // TODO: The provided `id` is sometimes hard-coded and used to interface with the
-            // component, e.g. `post_textbox`, so it can't be changed. This would ideally be
-            // abstracted to avoid passing in an `id` prop at all, but we intentionally maintain
-            // the old behaviour to address ABC-213.
-            id = 'autosize_textarea',
-            ...otherProps
-        } = props;
-
-        const heightProps = {
-            rows: 0,
-            height: 0,
-        };
-
-        Reflect.deleteProperty(otherProps, 'onWidthChange');
-
-        if (this.height <= 0) {
-            // Set an initial number of rows so that the textarea doesn't appear too large when its first rendered
-            heightProps.rows = 1;
-        } else {
-            heightProps.height = this.height;
-        }
+    if (height.current <= 0) {
+        // Set an initial number of rows so that the textarea doesn't appear too large when its first rendered
+        heightProps.rows = 1;
+    } else {
+        heightProps.height = height.current;
+    }
 
-        let textareaPlaceholder = null;
-        const placeholderAriaLabel = placeholder ? placeholder.toLowerCase() : '';
-        if (!this.props.value && !this.props.defaultValue) {
-            textareaPlaceholder = (
-                <div
-                    {...otherProps as any}
-                    id={`${id}_placeholder`}
-                    data-testid={`${id}_placeholder`}
-                    style={styles.placeholder}
-                >
-                    {placeholder}
-                </div>
-            );
-        }
+    let textareaPlaceholder = null;
+    const placeholderAriaLabel = placeholder ? placeholder.toLowerCase() : '';
+    if (!value && !defaultValue) {
+        textareaPlaceholder = (
+            <div
+                id={`${id}_placeholder`}
+                data-testid={`${id}_placeholder`}
+                style={styles.placeholder}
+            >
+                {placeholder}
+            </div>
+        );
+    }
 
-        return (
-            <div>
-                {textareaPlaceholder}
+    return (
+        <div>
+            {textareaPlaceholder}
+            <textarea
+                ref={setTextareaRef}
+                data-testid={id}
+                id={id}
+                {...heightProps}
+                role='textbox'
+                aria-label={placeholderAriaLabel}
+                dir='auto'
+                disabled={disabled}
+                onChange={onChange}
+                onInput={onInput}
+                value={value}
+                defaultValue={defaultValue}
+            />
+            <div style={styles.container}>
                 <textarea
-                    ref={this.setTextareaRef}
-                    data-testid={id}
-                    id={id}
-                    {...heightProps}
-                    {...otherProps}
```

Comment:
`otherProps` removed here because the only other props should be `onChange`, which is already passed.